### PR TITLE
libphidget22: init at 0-unstable-2024-04-11

### LINKED ISSUE #335946: Package request: Phidget
**Project description**

Phidgets are building-blocks for sensing and control using a computer, tablet, or phone. Phidgets enable your software application to interact with the physical world.

**Metadata**

* homepage URL: https://www.phidgets.com/
* source URL: 
 - https://www.phidgets.com/docs/OS_-_macOS
 - https://www.phidgets.com/docs/OS_-_Linux

* license: LGPL-3.0-or-later and BSD-2-Clause and BSD-3-Clause
* platforms: Linux, Darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by mksafavi**
## Description of changes
closes https://github.com/NixOS/nixpkgs/issues/335946

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/li/libphidget22extra/package.nix
Comment (donovanglover) [inline pkgs/by-name/li/libphidget22extra/package.nix:1]: Same suggestions apply to this file.

File: pkgs/by-name/li/libphidget22/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  stdenv,
+  fetchurl,
+  automake,
+  libusb1,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libphidget22";
+  version = "0-unstable-2024-04-11";
+  src = fetchurl {
```
Comment (donovanglover) [inline pkgs/by-name/li/libphidget22/package.nix:12]:
```suggestion
  version = "0-unstable-2024-04-11";

  src = fetchurl {
```
Nit: Convention is to put an empty line between `version` and `src`

**Comment by mksafavi** [comment]
Hi. thanks for reviewing this.

I added the suggestions you made.

I also enabled strictDebs.

I think the original package requester wasn't sure about the license that's why they listed many of them. checking here [phidget license](https://www.phidgets.com/docs/Software_License#Linux) they only mention BSD3.

**Comment by mksafavi** [comment]
Hi.
Is there anything to be done for this to get merged?

**Review by drupol — APPROVED** [review]

**Comment by vcunat** [comment]
The hash doesn't match (anymore).  Now I'm getting
```
sha256-l8lwEpdR87U2pb0jOAkrI/157B+87QvSVtAtOfedaBo=
```

**Comment by drupol** [comment]
How are we supposed to deal with packages that doesn't have a unique URL per version ?

**Comment by vcunat** [comment]
Yeah, that's bad.  I don't know.  Someone could tell them that it's annoying.

**Comment by drupol** [comment]
@mksafavi Are you willing to help with this?

**Pull request merged**